File: TH3/B2/bai2.js
async function calculate() {
    const num1 = parseFloat(document.getElementById('num1').value);
    const num2 = parseFloat(document.getElementById('num2').value);

    if (isNaN(num1) || isNaN(num2)) {
        alert('Please enter valid numbers');
        return;
    }

    const operations = ['add', 'subtract', 'multiply', 'divide'];
    const results = {};

    for (let operation of operations) {
        const response = await fetch(`http://localhost:8000/${operation}?a=${num1}&b=${num2}`);
        const data = await response.json();
        results[operation] = data.result !== undefined ? data.result : data.error;
    }

    document.getElementById('sum').value = results.add;
    document.getElementById('difference').value = results.subtract;
    document.getElementById('product').value = results.multiply;
    document.getElementById('quotient').value = results.divide;
}

File: TH3/B2/bai2.py
from fastapi import FastAPI
from typing import Union

app = FastAPI()

@app.get("/add")
async def add(a: Union[int, float], b: Union[int, float]):
    return {"result": a + b}

@app.get("/subtract")
async def subtract(a: Union[int, float], b: Union[int, float]):
    return {"result": a - b}

@app.get("/multiply")
async def multiply(a: Union[int, float], b: Union[int, float]):
    return {"result": a * b}

@app.get("/divide")
async def divide(a: Union[int, float], b: Union[int, float]):
    if b == 0:
        return {"error": "Cannot divide by zero"}
    return {"result": a / b}

if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8000)
